packages/block-library/src/query-title/index.php

```diff
@@ -5,6 +5,29 @@
  * @package WordPress
  */
 
+/**
+ * Filters the prefix from the `core/query-title` block archive variation.
+ *
+ * @param array $title Archive title to be displayed.
+ *
+ * @return string Returns the query title based on the queried object but without prefix.
+ */
+function block_query_title_filter_archive_title( $title ) {
+	if ( is_category() ) {
+		$title = single_cat_title( '', false );
+	} elseif ( is_tag() ) {
+		$title = single_tag_title( '', false );
+	} elseif ( is_author() ) {
+		$title = get_the_author();
+	} elseif ( is_post_type_archive() ) {
+		$title = post_type_archive_title( '', false );
+	} elseif ( is_tax() ) {
+		$title = single_term_title( '', false );
+	}
```

Comment:
Nitpick: We don't really need to set a var here, we can just return the result in each of these conditions.
```suggestion
	if ( is_category() ) {
		return single_cat_title( '', false );
	}
	if ( is_tag() ) {
		return single_tag_title( '', false );
	}
	if ( is_author() ) {
		return get_the_author();
	}
	if ( is_post_type_archive() ) {
		return post_type_archive_title( '', false );
	}
	if ( is_tax() ) {
		return single_term_title( '', false );
	}
```